runtime-api/bonded-currency/src/pool_details.rs

```diff
@@ -0,0 +1,62 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use pallet_bonded_coins::{curves::Curve, PoolDetails};
+use parity_scale_codec::{alloc::string::String, Decode, Encode};
+use scale_info::TypeInfo;
+use sp_std::vec::Vec;
+
+/// Enum to represent the operation of minting or burning tokens.
+#[derive(Decode, Encode, TypeInfo)]
+pub enum Operation {
```

Comment:
As I mentioned in the other comment, I would simply define this as `Operation<Amount>` and use this type directly in the API definition, without making the `Operation` type generic.

Reply:
[f62d168](https://github.com/KILTprotocol/kilt-node/pull/828/commits/f62d1688bec11d4d1a5127bff412d23b21e89a3b)